tests/openapi/test_strictmode.py

```diff
@@ -461,3 +461,83 @@ def search_request():
     assert "upsert" in search_fail.json()['status']['error']
     assert not search_fail.ok
 
+
+def test_strict_mode_read_rate_limiting(collection_name):
+    set_strict_mode(collection_name, {
+        "enabled": True,
+        "read_rate_limit_per_sec": 1,
+    })
+
+    response = request_with_validation(
+        api='/collections/{collection_name}',
+        method="GET",
+        path_params={'collection_name': collection_name},
+    )
+
+    assert response.ok
+    new_strict_mode_config = response.json()['result']['config']['strict_mode_config']
+    assert new_strict_mode_config['enabled']
+    assert new_strict_mode_config['read_rate_limit_per_sec'] == 1
+
+    failed = False
+
+    for _ in range(10):
+        response = request_with_validation(
+            api='/collections/{collection_name}/points/search',
+            method="POST",
+            path_params={'collection_name': collection_name},
+            body={
+                "vector": [0.2, 0.1, 0.9, 0.7],
+                "limit": 4
+            }
+        )
+        if not response.ok:
+            assert response.status_code == 429
+            assert "Rate limiting exceeded: Read rate limit exceeded, retry later" in response.json()['status']['error']
+            failed = True
+            break
+
+    assert failed, "Rate limiting did not work"
+
+
+def test_strict_mode_write_rate_limiting(collection_name):
+    set_strict_mode(collection_name, {
+        "enabled": True,
+        "write_rate_limit_per_sec": 1,
+    })
+
+    response = request_with_validation(
+        api='/collections/{collection_name}',
+        method="GET",
+        path_params={'collection_name': collection_name},
+    )
+
+    assert response.ok
+    new_strict_mode_config = response.json()['result']['config']['strict_mode_config']
+    assert new_strict_mode_config['enabled']
+    assert new_strict_mode_config['write_rate_limit_per_sec'] == 1
+
+    failed = False
+
+    for _ in range(10):
+        response = request_with_validation(
+            api='/collections/{collection_name}/points',
+            method="PUT",
+            path_params={'collection_name': collection_name},
+            query_params={'wait': 'true'},
+            body={
+                "points": [
+                    {
+                        "id": 1,
+                        "vector": [0.05, 0.61, 0.76, 0.74],
+                    },
+                ]
+            }
+        )
+        if not response.ok:
+            assert response.status_code == 429
+            assert "Rate limiting exceeded: Write rate limit exceeded, retry later" in response.json()['status']['error']
+            failed = True
+            break
+
+    assert failed, "Rate limiting did not work"
```

Comment:
Shall we make it a bit more strict and make `failed` a counter? And assert we've failed at least 5 (50%) times?